standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -568,6 +568,7 @@ public OpenTxnsResponse openTxns(OpenTxnRequest rqst) throws MetaException {
            */
           LOG.error("OpenTxnTimeOut exceeded commit duration {}, deleting transactionIds: {}", elapsedMillis, txnIds);
           deleteInvalidOpenTransactions(dbConn, txnIds);
+          dbConn.commit();
```

Comment:
Why is this change?

Reply:
It was missing, the delete was not persisted. When I was testing manually, I didn't notice, because I was querying back the data on the same connection...
It does not cause a big problem, these transactions would be cleaned up eventually when they time out.

Reply:
I thought close(null, stmt, dbConn) commits, but you are right, that it does not commit.
Shall we create a test case for this? :)

Reply:
Maybe in a follow-up jira...